docs/README.md

```diff
@@ -0,0 +1,52 @@
+# GenZ Documentation
+
+> This documentation is a work in progress and will be updated as the project progresses.
+
+Welcome to the GenZ documentation. 
+
+## What is GenZ?
+
+GenZ is generic golang generator. It is a tool that can be used to generate code from templates.
+The templates are written in golang's template language by users. The templates can be used to generate any kind of code.
+
+## Why GenZ?
+
```

Comment:
### Problem statement

> [Clear is better than clever](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=14m35s) (Rob Pike)

Clear code is easier to understand and modify, which leads to fewer errors and less time spent debugging. This is essential in industry where projects are often long-term and involve multiple developers. It also improves communication and collaboration.

But we have observed that, sometimes, particularly on large data (domain oriented) models, simple code tasks (like declaring constructor, setters, check model invariants or provide tooling around a model) become a bit repetitive and [Toil](https://sre.google/sre-book/eliminating-toil/).

### Extend my models with declaration and generation

Some large projects like Kubernetes have abstracted boilerplate code by leveraging on static analysis and generation (e.g., the [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) generator). For example, adding Go comments on Go interfaces declaration helps the generator to generate code based on these comments.

#### Parenthesis on reflection
Reflection is one way to achieve code introspection and access your model attributes, types and methods, but it's [hard to manipulate and operates at runtime](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=15m22s), which is very unsafe.

**We wanted a simple, generic and testable solution to introspect our models and generate code at compile time.**

We've built GenZ as a simple tool to help eliminate developer [Toil](https://sre.google/sre-book/eliminating-toil/) tied to your models.